File: Java/ArrayRotation.java
/*
Given an array, rotate the array to the right by k steps, where k is non-negative.
Example:
        [1,2,3,4,5,6,7] , k = 3 => [5,6,7,1,2,3,4]
        [-1,-100,3,99], k = 2 => [3,99,-1,-100]
*/
class ArrayRotation {
    public void rotate(int[] nums, int k) {
        k %= nums.length;
        if (k == 0) return; // Rotated array will be same as original array
        reverse(0, nums.length-1, nums);
        reverse(0, k-1, nums);
        reverse(k, nums.length-1, nums);
    }
    
    private void reverse(int start, int end, int[] nums){        
        while(start < end){
            int temp = nums[start];
            nums[start++] = nums[end];
            nums[end--] = temp;
        }
    }
}

File: CPP/array rotation.cpp
#include <iostream>
using namespace std;


void reverseArray(int arr[],int start,int end)
{

    while(start<end)
    {

        int temp= arr[start];
        arr[start]=arr[end];
        arr[end]=temp;
        start++;
        end--;

    }
}

void printArray(int arr[],int size){
    for(int i=0;i<size;i++){
        cout<<arr[i];
    }
}

int main(){
    int arr[] = {1, 2, 3, 4, 5, 6};

    int n = sizeof(arr) / sizeof(arr[0]);

    // To print original array
    printArray(arr, n);

    // Function calling
    reverseArray(arr, 0, n-1);

    cout << "Reversed array is" << endl;

    // To print the Reversed array
    printArray(arr, n);

    return 0;








}



//to reverse string  cout << b.at(0) << a.substr(1) << ' ';
   // cout << a.at(0) << b.substr(1) << endl;






